src/box/txn.c

```diff
@@ -1280,6 +1280,8 @@ box_txn_commit(void)
 		diag_set(ClientError, ER_COMMIT_IN_SUB_STMT);
 		return -1;
 	}
+	if (txn_check_can_tcl(txn) != 0)
```

Comment:
Logically, it seems like this check should be before the substatement check. Also same issue in `box_txn_rollback` and `box_txn_rollback_to_savepoint`.

Reply:
AFAIU, transaction cannot be committed or rolled back if it's in a sub statement. So these check are actually independent.
But logically, as for me, using `rollback` or `commit` in sub statements is a more serious mistake, so it should be checked firstly.